packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/Models/Friend.Serialization.cs

```diff
@@ -43,6 +46,13 @@ protected virtual void JsonModelWriteCore(Utf8JsonWriter writer, ModelReaderWrit
             }
             writer.WritePropertyName("name"u8);
             writer.WriteStringValue(Name);
+            if (Optional.IsDefined(PropertyWithUnfriendlyWireName))
```

Comment:
Can we add these as unit tests instead of new cases in our test library?  I want to rely more on unit tests since they are easier to catch regressions vs noticing some change in a test generation.